doc/src/manual/style-guide.md

```diff
@@ -413,3 +413,54 @@ julia> h(1)
 
 Thus, use `Int` literals when possible, with `Rational{Int}` for literal non-integer numbers,
 in order to make it easier to use your code.
+
+## In long-form function definitions, prefer explicit `return` over implicit return values
+
+Explicit `return` on an expression makes it clear that the expression is intended to be returned and
+is not only used for its side effect. It also makes it possible to locally see that a certain sub-expression
+in a larger expression is returned while for an implicit return value one needs to analyze the whole outer expression.
+
```

Comment:
Maybe also mention that it can help the program to run faster if it was returning an unintentional value (when it only needs to return `nothing`)?

Reply:
Done